python/ray/serve/_private/logging_utils.py

```diff
@@ -183,6 +189,41 @@ def get_component_logger_file_path() -> Optional[str]:
                 return absolute_path[len(ray_logs_dir) :]
 
 
+class StreamToLogger(object):
```

Comment:
Can we add some tests for this thing? And how does it interact with JSON logging?

Reply:
This is currently tested through seeing the log in serve's log file (meaning that serve logger is used correctly). Will add some unit tests around each method. Mostly probably will just be testing the methods are called correctly 😅

Reply:
if there isn't utility in the tests then don't worry about it

Reply:
I added them https://github.com/ray-project/ray/pull/43508/files#diff-8f8bf3fd66946b2d713bf2c9ceddbeb2a056c30b9c5ae76cf034f6367ad96acfR708 XD